pkg/internal/telemetry/interfaces.go

```diff
@@ -0,0 +1,25 @@
+package telemetry
+
+import (
+	"github.com/prefab-cloud/prefab-cloud-go/pkg/internal/options"
+	prefabProto "github.com/prefab-cloud/prefab-cloud-go/proto"
+)
+
+type Aggregator interface {
+	Record(data interface{})
+	GetData() *prefabProto.TelemetryEvent
+	Clear()
+	Lock()
+	Unlock()
+}
+
+func NewContextAggregator(opts options.Options) Aggregator {
+	switch opts.ContextTelemetryMode {
+	case options.ContextTelemetryModes.PeriodicExample:
+		return NewExampleContextAggregator()
```

Comment:
it makes sense to not submit shapes and the examples, but currently I don't think the API extracts shapes from the example stream; plus I think this means we never learn the shape of no-key contexts
I think we should run the ShapeAggregator even when PeriodicExample is selected: thoughts?

Reply:
Interesting. I'm not sure about the Java client, but the Ruby client does EITHER shape OR example and not both. 

🤔 If you're submitting examples, the shapes don't tell us any more than you're already giving us, so I don't have a privacy problem here.

I guess that's a fine change. Should the other clients do likewise?

Reply:
i'm pretty sure the java client does this already, i'll have to check python. Short answer: yes

Reply:
also on shapes: the shape is pretty much finalized early in the lifetime of a client, so in at least some of my clients, i don't send shapes unless its marked dirty relative to what's been sent before.
The example contexts I also only send if they've changed on some time scale: sending the same eg server context every upload interval is just spending money with very low value

Reply:
For this PR, I'm going to make the change to ship the shapes as well as the examples.

I'm going to leave the optimizations you mentioned as a future task. I'll capture this in Linear.

Reply:
Shapes added to periodic examples in b7753a8